gyp/pylib/gyp/input.py

```diff
@@ -961,33 +961,26 @@ def ExpandVariables(input, phase, variables, build_file):
                     # Fix up command with platform specific workarounds.
                     contents = FixupPlatformCommand(contents)
                     try:
-                        p = subprocess.Popen(
+                        # stderr will be printed no matter what
+                        result = subprocess.run(
                             contents,
-                            shell=use_shell,
                             stdout=subprocess.PIPE,
```

Comment:
```suggestion
```

Reply:
`stdout=subprocess.PIPE,` is needed for `result.stdout` to exist

Reply:
Only if `text = True` is absent?

Reply:
test this:

Fails:
```python
import subprocess
result = subprocess.run(
    'node -p "3245"',
    check=False,
    text=True
)
print(result.stdout) # None
```

Succeeds:
```python
import subprocess
result = subprocess.run(
    'node -p "3245"',
    stdout=subprocess.PIPE,
    check=False,
    text=True
)
print(result.stdout) # 3245
```

Reply:
Perfect.  Thanks for testing that out.  I think this is looking good.